tests/integration/aof-multi-part.tcl

```diff
@@ -1166,5 +1166,138 @@ tags {"external:skip"} {
+            catch {exec rm -rf $aof_dirpath}
+            r debug loadaof
+
+            assert_equal 0 [r dbsize]
+
+            # Increase AOFRW execution time to give us enough time to kill it
+            r config set rdb-key-save-delay 10000000
+
+            # Start write load
+            set load_handle0 [start_write_load $master_host $master_port 10]
+
+            wait_for_condition 50 100 {
+                [r dbsize] > 0
+            } else {
+                fail "No write load detected."
+            }
+
+            r config set appendonly yes
+            
+            # Make sure AOFRW was scheduled 5 times
+            set total_forks [s total_forks]
+            set maxtries 1000
+            set delay 10
+            while {[incr maxtries -1] >= 0} {
+                if {[s total_forks] == [expr $total_forks + 5]} {
+                    break
+                }
+                catch {exec kill -9 [get_child_pid 0]}
+                after $delay
+            }
+            if {$maxtries == -1} {
+                fail "aof rewrite did not scheduled 5 times"
+            }
+           
+            # Make sure manifest file is not created
+            assert_equal 0 [check_file_exist $aof_dirpath $aof_manifest_name]
+            # Make sure BASE AOF is not created
+            assert_equal 0 [check_file_exist $aof_dirpath "${aof_basename}.1${::base_aof_sufix}${::rdb_format_suffix}"]
+            # Make sure temp INCR AOF is created
+            assert_equal 1 [check_file_exist $aof_dirpath "${aof_basename}.1${::incr_aof_sufix}${::aof_format_suffix}"]
+
+            # Do a successful AOFRW
+            set total_forks [s total_forks]
+            r config set rdb-key-save-delay 0
+            catch {exec kill -9 [get_child_pid 0]}
+            
+            wait_for_condition 1000 10 {
+                [s total_forks] == [expr $total_forks + 1]
+            } else {
+                fail "aof rewrite did not scheduled"
+            }
+            waitForBgrewriteaof r
+
+            # BASE and INCR AOF are successfully created
+            assert_aof_manifest_content $aof_manifest_file {
+                {file appendonly.aof.1.base.rdb seq 1 type b}
+                {file appendonly.aof.1.incr.aof seq 1 type i}
+            }
+
+            stop_write_load $load_handle0
+            wait_load_handlers_disconnected
+
+            set d1 [r debug digest]
+            r debug loadaof
+            set d2 [r debug digest]
+            assert {$d1 eq $d2}
+
+            # Dynamic disable AOF again
+            r config set appendonly no
+
+            # Disabling AOF does not delete previous AOF files
+            r debug loadaof
+            set d2 [r debug digest]
+            assert {$d1 eq $d2}
+
+            r config set rdb-key-save-delay 10000000
+            set load_handle0 [start_write_load $master_host $master_port 10]
+
+            # Re-enable AOF
+            r config set appendonly yes
```

Comment:
let's make sure there's no race here leading to some rare scenario and wait for seeing the load handler running before we enable appendonly.
we can do that by monitoring the command stats, or rdb_changes_since_last_save (since the dbsize is hard to trust now)

Reply:
ohh, `rdb_changes_since_last_save` looks can work.